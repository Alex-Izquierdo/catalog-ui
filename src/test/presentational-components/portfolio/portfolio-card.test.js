import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PortfolioCard from '../../../PresentationalComponents/Portfolio/PorfolioCard';

describe('<PortfolioCard />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      portfolioName: 'foo',
      description: 'description',
      modified: 'modified',
      name: 'name',
      id: '123'
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<PortfolioCard { ...initialProps } />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render and create correct card actions', () => {
    const wrapper = mount(
      <MemoryRouter>
        <PortfolioCard { ...initialProps } />
      </MemoryRouter>
    );
    expect(wrapper.find('button')).toHaveLength(2);

  });
});
