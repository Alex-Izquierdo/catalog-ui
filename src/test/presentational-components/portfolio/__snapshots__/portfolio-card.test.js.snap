// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioCard /> should render correctly 1`] = `
<GalleryItem>
  <Card
    className="content-gallery-card"
    component="article"
  >
    <Link
      className="card-link"
      to="/portfolios/detail/123"
    >
      <CardHeader
        className="card-image-header"
        component="div"
      >
        <PortfolioCardHeader
          backgroundSrc="test-file-stub"
          headerActions={
            Array [
              <Dropdown
                className=""
                direction="down"
                dropdownItems={
                  Array [
                    <Item
                      className=""
                      component="a"
                      href="#"
                      isDisabled={false}
                      isHovered={false}
                    >
                      <Link
                        className="pf-c-dropdown__menu-item"
                        to="/portfolios/edit/123"
                      >
                        Edit
                      </Link>
                    </Item>,
                    <Item
                      className=""
                      component="a"
                      href="#"
                      isDisabled={false}
                      isHovered={false}
                    >
                      <Link
                        className="pf-c-dropdown__menu-item destructive-color"
                        to="/portfolios/remove/123"
                      >
                        Delete
                      </Link>
                    </Item>,
                  ]
                }
                isOpen={false}
                isPlain={true}
                onSelect={[Function]}
                position="right"
                toggle={
                  <Kebab
                    aria-label="Actions"
                    className=""
                    id=""
                    isActive={false}
                    isFocused={false}
                    isHovered={false}
                    isOpen={false}
                    isPlain={false}
                    onToggle={[Function]}
                    parentRef={null}
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  />
                }
              />,
            ]
          }
          portfolioName="name"
        />
      </CardHeader>
      <CardBody
        className=""
        component="div"
      >
        <TextContent
          className=""
        >
          <Text
            className=""
            component="small"
          >
            Last modified 
            Invalid Date
             by 
          </Text>
        </TextContent>
        <ItemDetails
          description="description"
          modified="modified"
          name="name"
          portfolioName="foo"
          toDisplay={
            Array [
              "description",
            ]
          }
        />
      </CardBody>
      <CardFooter
        className=""
        component="div"
      />
    </Link>
  </Card>
</GalleryItem>
`;
